server/odc-migrate/src/main/resources/migrate/common/V_4_2_4_1__alter_notification_schema.sql

```diff
@@ -0,0 +1,104 @@
+DROP TABLE IF EXISTS `notification_channel`;
+CREATE TABLE `notification_channel` (
+  `id` bigint(20) NOT NULL AUTO_INCREMENT,
+  `create_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `update_time` datetime DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  `creator_id` bigint(20) NOT NULL COMMENT 'creator user id, references iam_user.id',
+  `organization_id` bigint(20) NOT NULL COMMENT 'organization id, references iam_organization.id',
+  `name` varchar(128) NOT NULL COMMENT 'channel name',
+  `type` varchar(128) NOT NULL COMMENT 'channel type, may DingTalk, SMS, etc.',
+  `project_id` bigint(20) NOT NULL COMMENT 'project id, references collaboration_project.id',
+  `description` varchar(512) DEFAULT NULL COMMENT 'description',
+  PRIMARY KEY (`id`),
+  KEY `notification_channel_project_id` (`project_id`)
+) COMMENT = 'notification channel configs';
+
+DROP TABLE IF EXISTS `notification_policy`;
+CREATE TABLE IF NOT EXISTS `notification_policy`(
+  `id` bigint(20) NOT NULL AUTO_INCREMENT,
+  `create_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `update_time` datetime DEFAULT CURRENT_TIMESTAMP ON update CURRENT_TIMESTAMP,
+  `creator_id` bigint(20) NOT NULL comment 'creator user id, references iam_user.id',
+  `organization_id` bigint(20) NOT NULL comment 'organization id, references iam_organization.id',
+  `title_template` text DEFAULT NULL comment 'notification message title template, which could contains variables',
+  `content_template` text DEFAULT NULL comment 'notification message content template, which could contains variables',
+  `match_expression` varchar(2048) NOT NULL comment 'indicate if a event matches the expression',
+  `to_users` varchar(2048) DEFAULT NULL comment 'odc users who will receive this message',
+  `cc_users` varchar(2048) DEFAULT NULL comment 'odc users who will receive this message by copy',
+  `is_enabled` tinyint(1) NOT NULL DEFAULT '1' COMMENT 'Flag bit, mark whether the triggered rule is enabled',
+  `project_id` bigint(20) NOT NULL COMMENT 'project id, references collaboration_project.id',
+  `policy_metadata_id` bigint(20) NOT NULL COMMENT 'notification policy metadata id, reference to notification_policy_metadata.id',
+  CONSTRAINT `pk_notification_policy_id` PRIMARY KEY (`id`),
+  KEY `notification_policy_project_id` (`project_id`)
+) COMMENT = 'notification policy';
+
+CREATE TABLE IF NOT EXISTS `notification_policy_metadata`(
+  `id` bigint(20) NOT NULL AUTO_INCREMENT,
+  `create_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `update_time` datetime DEFAULT CURRENT_TIMESTAMP ON update CURRENT_TIMESTAMP,
+  `event_category` varchar(128) NOT NULL comment 'notification event category, such as TASK',
+  `event_name` varchar(128) NOT NULL comment 'notification event name',
+  `match_expression` varchar(2048) comment 'indicate if a event matches the expression',
```

Comment:
notification_policy also has a column `match_expression`, are them duplicated?

Reply:
The `metch_expression` column in `notification_policy` is generated from this metadata table~  So they are not duplicated.

Reply:
> The `metch_expression` column in `notification_policy` is generated from this metadata table~ So they are not duplicated.

I didn't understand. How is it generated?